lib/parser.ts

```diff
@@ -1,35 +1,109 @@
-import { Account, RawAccount, AccountStatus } from './entities'
+import {
+  Account, RawAccount, AccountStatus, RawOrder, OrderType, OrderSide, Order,
+  OrderTimeInForce, OrderStatus, RawPosition, Position, PositionSide
+} from './entities'
 
 export class Parser {
   parseAccount(rawAccount: RawAccount): Account {
     if (!rawAccount) {
       return null
     }
 
-    return {
-      ...rawAccount,
-      buying_power: this.parseNumber(rawAccount.buying_power),
-      regt_buying_power: this.parseNumber(rawAccount.regt_buying_power),
-      daytrading_buying_power: this.parseNumber(
-        rawAccount.daytrading_buying_power
-      ),
-      cash: this.parseNumber(rawAccount.cash),
-      portfolio_value: this.parseNumber(rawAccount.portfolio_value),
-      multiplier: this.parseNumber(rawAccount.multiplier),
-      equity: this.parseNumber(rawAccount.equity),
-      last_equity: this.parseNumber(rawAccount.last_equity),
-      long_market_value: this.parseNumber(rawAccount.long_market_value),
-      short_market_value: this.parseNumber(rawAccount.short_market_value),
-      initial_margin: this.parseNumber(rawAccount.initial_margin),
-      maintenance_margin: this.parseNumber(rawAccount.maintenance_margin),
-      last_maintenance_margin: this.parseNumber(
-        rawAccount.last_maintenance_margin
-      ),
-      sma: this.parseNumber(rawAccount.sma),
-      status: rawAccount.status as AccountStatus,
+    try {
+      return {
+        ...rawAccount,
+        buying_power: this.parseNumber(rawAccount.buying_power),
+        regt_buying_power: this.parseNumber(rawAccount.regt_buying_power),
+        daytrading_buying_power: this.parseNumber(
+          rawAccount.daytrading_buying_power
+        ),
+        cash: this.parseNumber(rawAccount.cash),
+        portfolio_value: this.parseNumber(rawAccount.portfolio_value),
+        multiplier: this.parseNumber(rawAccount.multiplier),
+        equity: this.parseNumber(rawAccount.equity),
+        last_equity: this.parseNumber(rawAccount.last_equity),
+        long_market_value: this.parseNumber(rawAccount.long_market_value),
+        short_market_value: this.parseNumber(rawAccount.short_market_value),
+        initial_margin: this.parseNumber(rawAccount.initial_margin),
+        maintenance_margin: this.parseNumber(rawAccount.maintenance_margin),
+        last_maintenance_margin: this.parseNumber(
+          rawAccount.last_maintenance_margin
+        ),
+        sma: this.parseNumber(rawAccount.sma),
+        status: rawAccount.status as AccountStatus,
+      }
+    } catch (err) {
+      throw new Error(`Account parsing failed. Error: ${err.message}`)
```

Comment:
@117 Here's the basic error handling approach. Just try/catch the entire parse step and give a helpful error message. Let me know what you think.

Reply:
this is exactly what i had in mind, provides context and the error